### ISSUE TITLE
[Bug] from __future__ import annotations

### ISSUE THREAD

**Issue by jmolinski** [ramonhagenaars/jsons#26]
Hello!

`from __future__ import annotations` literally destroys the library. 

Quick example:

`foo.py`
```python
from __future__ import annotations

from dataclasses import dataclass


@dataclass
class Spam:
    eggs: int


@dataclass
class Baz:
    qux: int
    spam: Spam
```

`bar.py`
```python
from typing import List

import jsons

from foo import Baz

data = [{"qux": 0, "spam": {"eggs": 1}}]

x = jsons.load(data, List[Baz])
```

Expected result: `[Baz(qux=0, spam=Spam(eggs=1))]`
Result: 
```
>>> import bar
Traceback (most recent call last):
  File "/home/glen/traktpy/venv/lib/python3.7/site-packages/jsons/_main_impl.py", line 146, in load
    return deserializer(json_obj, cls, **kwargs_)
TypeError: 'NoneType' object is not callable

During handling of the above exception, another exception occurred:

Traceback (most recent call last):

...

  File "/home/glen/traktpy/venv/lib/python3.7/site-packages/jsons/_main_impl.py", line 150, in load
    raise DeserializationError(str(err), json_obj, cls)
jsons.exceptions.DeserializationError: 'NoneType' object is not callable
>>> 
```

If you remove `from __future__ import annotations` from `foo.py` everything works as expected.

https://www.python.org/dev/peps/pep-0563/
`from __future__ import annotations` stops the interpreter from executing type annotations, they are stored as str instead. Types may be revealed during runtime using `typing.get_type_hints()`. This is going to be the default behavior in python4.

**Comment by ramonhagenaars** [comment]
Hello jmolinsky. This looks serious, thank you for reporting it.

**Comment by ramonhagenaars** [comment]
This has been solved in [0.8.4](https://pypi.org/project/jsons/).
